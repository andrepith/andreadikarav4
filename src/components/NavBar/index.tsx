import { useState } from "react";
import Link from "next/link";
// import { List } from "react-bootstrap-icons";
import Overlay from "../Overlay";
import bioInterface from "src/lib/Types";

interface NavBarInterface extends bioInterface {
  scrollToExperience: () => void;
  scrollToProject: () => void;
  scrollToSkills: () => void;
  scrollToContact: () => void;
}

const NavBar = ({
  bio: { firstName, lastName, email },
  scrollToExperience,
  scrollToProject,
  scrollToSkills,
  scrollToContact,
}: NavBarInterface) => {
  const [isOpen, setOpen] = useState(false);
  const handleOpen = () => setOpen(true);
  const handleClose = () => setOpen(false);
  return (
    <>
      <Overlay
        isOpen={isOpen}
        handleClose={handleClose}
        scrollToExperience={scrollToExperience}
        scrollToProject={scrollToProject}
        scrollToSkills={scrollToSkills}
        scrollToContact={scrollToContact}
      />
      <header className="fixed-top navigation" id="navigation">
        <nav className="navbar navbar-expand-lg primary-nav container">
          <Link href="/">
            <a aria-current="page" className="nav-item nav-link pl-0 me-4">
              {firstName} {lastName}
            </a>
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            onClick={handleOpen}
            aria-label="nav-mobile"
          >
            {/* <List size={24} color="white" /> */}
          </button>
          <div className="collapse navbar-collapse" id="navbar">
            <div className="navbar-nav align-items-center ml-auto">
              <div
                onClick={scrollToExperience}
                className="nav-item nav-link me-2"
              >
                Experience
              </div>
              <div onClick={scrollToProject} className="nav-item nav-link me-2">
                Project
              </div>
              <div onClick={scrollToSkills} className="nav-item nav-link me-2">
                Skills
              </div>
              <div onClick={scrollToContact} className="nav-item nav-link me-2">
                Contact
              </div>
              <a
                href={`mailto:${email}`}
                className="btn btn-layered offset"
                role="button"
                aria-pressed="true"
              >
                hire me
              </a>
            </div>
          </div>
        </nav>
      </header>
    </>
  );
};

export default NavBar;
