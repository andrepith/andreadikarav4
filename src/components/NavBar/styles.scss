.navigation {
  z-index: 1000;
  position: fixed;
  background: rgba(122, 97, 27, 0.178);
  box-shadow: 0 8px 32px 0 rgb(31 38 135 / 2%);
  backdrop-filter: blur(10px);
  -webkit-backdrop-filter: blur(7px);
  border: 1px solid rgba(255, 255, 255, 0.02);
}

.navbar-toggler:focus {
  box-shadow: none;
}

.navbar-nav {
  .nav-link {
    font-size: 14px;
    font-weight: 500;
    cursor: pointer;
    transition: all 0.25s cubic-bezier(0.645, 0.045, 0.355, 1);

    &:hover {
      transform: translateY(-3px);
      color: $color-highlight;
    }
  }
}

.nav-link {
  color: $color-white;
  font-weight: 700;
  font-size: 1.25rem;
}

.primary-nav {
  flex-wrap: wrap;
  justify-content: space-between;
}

.primary-nav .navbar-collapse {
  flex-grow: unset;
}

.btn-layered {
  background: $color-primary;
  font-family: CircularStd, sans-serif !important;
  font-size: 10pt;
  font-weight: 900;
  line-height: 1;
  padding: 10px 26px;
  transition: 0.1s ease-out;
  -webkit-transition: 0.1s ease-out;
  color: $color-white;
}

.offset:focus,
.offset:hover {
  color: $color-white;
  text-decoration: none;
}

.nav-overlay {
  height: 100%;
  display: none;
  width: 100%;
  position: fixed;
  left: 0;
  top: 0;
  opacity: 0.95;
  background-color: $color-primary;
  overflow-x: hidden;
  -webkit-transition: 0.5s;
  transition: 0.5s;
  z-index: 999999 !important;
}

.nav-overlay .nav-close {
  position: absolute;
  top: 20px;
  right: 45px;
  font-size: 24pt;
  color: $color-white;
  cursor: pointer;
}

.overlay-content {
  position: relative;
  top: 25%;
  width: 100%;
  text-align: center;
  margin-top: 30px;
}

.nav-overlay div {
  padding: 8px;
  text-decoration: none;
  font-size: 36px;
  color: $color-white !important;
  display: block;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  font-weight: 700;
}
