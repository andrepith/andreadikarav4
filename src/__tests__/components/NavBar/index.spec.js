import renderer from "react-test-renderer";
import NavBar from "../../../components/NavBar";

it("Renders correctly", () => {
  const tree = renderer.create(<NavBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
