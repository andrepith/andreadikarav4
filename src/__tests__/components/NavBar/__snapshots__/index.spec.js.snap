// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Array [
  <aside
    className="nav-overlay d-none"
    id="navOverlay"
  >
    <span
      className="nav-close"
    >
      <svg
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
        />
        <path
          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
        />
      </svg>
    </span>
    <div
      className="overlay-content"
    >
      <div
        onClick={[Function]}
      >
        Experience
      </div>
      <div
        onClick={[Function]}
      >
        Project
      </div>
      <div
        onClick={[Function]}
      >
        Skills
      </div>
      <div
        onClick={[Function]}
      >
        Contact
      </div>
    </div>
  </aside>,
  <header
    className="fixed-top navigation"
    id="navigation"
  >
    <nav
      className="navbar navbar-expand-lg primary-nav container"
    >
      <a
        aria-current="page"
        className="nav-item nav-link pl-0 me-4"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
         
      </a>
      <button
        aria-label="nav-mobile"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="white"
          height={24}
          viewBox="0 0 16 16"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.5 12a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5z"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbar"
      >
        <div
          className="navbar-nav align-items-center ml-auto"
        >
          <div
            className="nav-item nav-link me-2"
          >
            Experience
          </div>
          <div
            className="nav-item nav-link me-2"
          >
            Project
          </div>
          <div
            className="nav-item nav-link me-2"
          >
            Skills
          </div>
          <div
            className="nav-item nav-link me-2"
          >
            Contact
          </div>
          <a
            aria-pressed="true"
            className="btn btn-layered offset"
            href="mailto:undefined"
            role="button"
          >
            hire me
          </a>
        </div>
      </div>
    </nav>
  </header>,
]
`;
